guacamole-common/src/main/java/org/apache/guacamole/GuacamoleAuthenticationRejectedException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.guacamole;
+
+/**
+ * An exception thrown when a successful authentication is rejected by a
+ * AuthenticationSuccessListener in an extension.
+ */
+public class GuacamoleAuthenticationRejectedException
+    extends GuacamoleSecurityException {
+
+    public GuacamoleAuthenticationRejectedException() {
```

Comment:
I haven't yet checked whether these exceptions make sense as API changes overall, but beware that the constructors of all classes need to be documented with javadoc, including exceptions.

It is also standard convention to provide the message, message/cause, and cause constructors. It would be odd if these exceptions are the only exceptions in the Guacamole API which deviate from that convention, instead providing their own default message.

See:

https://github.com/apache/incubator-guacamole-client/blob/a5ebc07349bcf43c80f404836d9964ac894bed26/guacamole-common/src/main/java/org/apache/guacamole/GuacamoleResourceClosedException.java#L30-L62